pgmpy/models/DynamicBayesianNetwork.py

```diff
@@ -195,7 +199,10 @@ def get_intra_edges(self):
         >>> dbn.get_intra_edges()
         [(('D', 0), ('G', 0)), (('G', 0), ('L', 0)), (('I', 0), ('G', 0))
         """
-        return [edge for edge in self.edges() if edge[0][1] == edge[1][1]]
+        if not str(time_slice).isdigit():
```

Comment:
@palashahuja What is this? A better way to do this check would be `if not isinstance(time_slice, int) or time_slice < 0`